### DISCUSSION TITLE
Multiple Waveshare Client/Servers

### DISCUSSION THREAD

**Discussion by andygodber** [alienatedsec/solis-ha-modbus-cloud#30] [Q&A] [answered]
I have a Solis inverter and an existing Waveshare RS485 to Eth (actually over powerline) to enable connection of the mains meter CT (which is not directly connectable). This is working.
My initial attempt to introduce this configuration (modbus-cloud) caused the inverter to error inidcating the meter connection was not working. Removing the new Waveshares, rebooting the inverter and all working.

I have updated the new waveshares firmware and configured per instructions.
I have not updated the original waveshare firmware nor changed the operating mode (Modbus Gateway type_ - both are in Simple Modbus TCP to RTU. 

Do I need to make any changes on the original (or new) Waveshares to allow compatibility?

Im guessing there's some potential conflict by either having a couple of servers (although surely the direction to named IP addresses should overcome this?) or the original waveshares operating in simple mode?

**Comment by alienatedsec** [comment]
Where is your Waveshare/Meter connected at the moment? Which port?

**Reply by andygodber** [reply]
The original server/client is ip xxx.200 and 201 both on the default port of 4196.
The new server/client is 202 and 203, again with the default of port 4196. Although I see some configs saying this should be 502.

**Reply by alienatedsec** [reply]
I meant where it is physically connected.:)

**Reply by andygodber** [reply, marked as answer]
lol. The exisiting is plugged into the Meter CT  port, through the waveshare1, over powerline adapter 1, through the mains, powerline adapter 2, waveshare2 to rs485 power meter with CT clamp at the mains import location.
New is wifi module port, through waveshare3, through lan to wavershare4 and wifi data logger out of that. Exceedcom male and female adapters appropriately.
I can see the wifi datalogger is working - it is reporting to Soliscloud.
But Im not seeing any entities created by the Solax HA integration.
I have now set the new waveshare ports to 502.

Edit: seeing this in HA log -
unrecognized solisinv inverter type - serial number : unknown
13:22:33 – (ERROR) SolaX Inverter Modbus (custom integration)
solisinv: cannot find serial number, even not for other Inverter
13:22:33 – (ERROR) SolaX Inverter Modbus (custom integration)
solisinv: reading serial number from address 0x80ec failed; other address may succeed
13:22:33 – (WARNING) SolaX Inverter Modbus (custom integration)

Edit #2 - ignore the HA problems - its now working! I had to change the HA integration to Modbus RTU over TCP, away from the default Modbus TCP

So just monitoring to see how offten the meter error is triggered.

**Reply by andygodber** [reply]
I think my CT Meter errors were coincidental - they've settled down to approx 1 per day, which is what Id observed over the last few months. Maybe connecting the new setup had disturbed something. So confirming possible to have multiple client/server config.
